### PR TITLE
drivers: spi: Add MEC172x QMSPI-LDMA driver

### PR DESCRIPTION
**Pull request by jvasanth1**
Add driver for MEC172x QMSPI with local DMA(LDMA). The driver
support SPI asynchronous operation.

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

### CODE REVIEW

**Review by jackrosenthal — APPROVED** [review]

**Review by albertofloyd — APPROVED** [review]

File: dts/bindings/spi/microchip,xec-qmspi-ldma.yaml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+    lines:
+      type: int
+      required: false
+      description: |
+        QMSPI data lines 1, 2, or 4. 1 data line is full-duplex
+        MOSI and MISO or half-duplex on MOSI only. Lines set to 2
+        or 4 indicate dual or quad I/O modes. Defaults to 1.
+
+    port_sel:
```
Comment (mbolivar-nordic) [inline dts/bindings/spi/microchip,xec-qmspi-ldma.yaml:39]: Please use dashes (`-`), not underscores (`_`) for all properties

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
One request for the binding.

I'm off next week, feel free to dismiss this review once the request is addressed

File: dts/bindings/spi/microchip,xec-qmspi-ldma.yaml (hunk 1 above)
Comment (jvasanth1) [reply dts/bindings/spi/microchip,xec-qmspi-ldma.yaml]: Fixed with latest push

**Comment by jvasanth1** [comment]
Hi Marti @mbolivar-nordic  I have updated based on your feedback. Could you please approve the merging?
Thanks.

File: dts/bindings/spi/microchip,xec-qmspi-ldma.yaml (hunk 1 above)
Comment (albertofloyd) [reply dts/bindings/spi/microchip,xec-qmspi-ldma.yaml]: @mbolivar-nordic dismissing the review to proceed with merge as suggested

**Review by mbolivar-nordic — APPROVED** [review]

**Comment by cfriedt** [comment]
@jvasanth1 - please rebase and fix merge conflicts
@tbursztyka - PTAL

**Comment by jvasanth1** [comment]
@cfriedt - Rebased and fixed merge conflicts

**Pull request merged**